src/Microsoft.Azure.WebJobs.Host/Scale/ScaleLoggerExtension.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    public static class ScaleLoggerExtension
+    {
+        private static readonly EventId FunctionScaleErrorEventId = new EventId(8001, "FunctionScaleError");
+        private static readonly EventId LogFunctionScaleVoteEventId = new EventId(8002, "LogFunctionScaleVote");
+        private static readonly string FunctionNameKey = "functionName";
+        private static readonly string TargetWorkerCountKey = "targetWorkerCount";
+        private static readonly string QueueLengthKey = "queueLength";
+        private static readonly string ConcurrencyKey = "concurrency";
+        private static readonly string VoteKey = "vote";
+        private static readonly string ReasonKey = "reason"; // Renamed from 'Reason'
+
+        // <summary>
+        // Logs a scale vote for a function with the specified name, vote, and optional reason.
+        // </summary>
+        public static void LogFunctionScaleVote(this ILogger logger, string functionName, string vote, string reason = null)
```

Comment:
Why is this added to public API? I am not familiar with scale controller APIs, but if this is expected for webjobs extension packages to call this, could we instead have them return these values as part of the API and then scale controller crafts the log from that?

Reply:
Not a fan of this approach, as it is a bit obscure for extensions to understand how to use this log API independently vs it being a natural part of existing API.

But won't hold up this PR on that. We can revisit this approach later.